pkg/webreg/webreg.go

```diff
@@ -1228,6 +1229,138 @@ const imageExampleLiteral = `- as: repo-e2e
           memory: 2Gi
 `
 
+const quotasAndLeasesPage = `<h2 id="title"><a href="#title">How are Cloud Quota and Aggregate Concurrency Limits Handled?</a></h2>
+<p>
+Jobs that interact with an Infrastructure-as-a-Service (IaaS) cloud provider use credentials shared across the broader
+CI platform. Therefore, all jobs interacting with a specific IaaS will use API quota for these cloud providers from a
+shared pool. In order to ensure that our job throughput for a provider remains within the aggregate limit imposed by
+shared quota, jobs acquire leases for slices of the quota before they run and only relinquish them once all actions are
+completed. This document will describe the mechanism used to provide leases of quota slices to jobs, how jobs determine
+which quota to ask for, how available leases can be configured and how current usage can be investigated.
+</p>
+
+<h3 id="boskos"><a href="#boskos">Introducing the <code>boskos</code> Leasing Server</a></h3>
+<p>
+<code>boskos</code> (βοσκός), translating as "shepherd" from Greek, is a resource management server that apportions
+<i>leases</i> of <i>resources</i> to clients and manages the lifecycle of the <i>resources</i>. When considering the
+actions of this server, two terms should be defined:
+</p>
+
+<table class="table">
+  <tr>
+    <th style="white-space: nowrap">Term</th>
+    <th>Definition</th>
+  </tr>
+  <tr>
+    <td style="white-space: nowrap"><i>resource</i></td>
+    <td>An item which may be leased to clients. In the context of CI jobs, resources are slices of the larger cloud quota.</td>
+  </tr>
+  <tr>
+    <td style="white-space: nowrap"><i>lease</i></td>
+    <td>A binding between a <i>resource</i> and a client. When a lease is active, the underlying <i>resource</i> is not available for other clients.</td>
+  </tr>
+</table>
+
+<p>
+The process for granting a <i>lease</i> on a <i>resource</i> from the available pool generally follows this workflow:
+</p>
+
+<ul>
+  <li>a client (<i>lessee</i>) requests a <i>lease</i> on an available <i>resource</i></li>
+  <li>the server (<i>lessor</i>) grants the <i>lease</i>, if possible, or places the client in a FIFO queue to wait for the next available <i>resource</i></li>
+  <li>the client emits a heartbeat while the <i>lease</i> is under active use</li>
+  <li>the client relinquishes the <i>lease</i> once the it is no longer in use</li>
+  <li>the server places the <i>resource</i> into the available pool for future clients to request</li>
+</ul>
+
+<p>
+If a client fails to emit a heartbeat for long enough while the client holds a <i>lease</i>, the server will forcibly
+relinquish the <i>lease</i> and return the <i>resource</i> to the available pool for other clients. This mechanism
+ensures that clients which crash or otherwise fail to remain responsive cannot exhaust <i>resources</i> by holding a
+<i>lease</i> indefinitely.
```

Comment:
unrelated to the doc, but can ci-op properly cope with the case that its resource is gone, e.G. because it got starved or sth? Or do we workaround that by having high timeouts?

Reply:
So you mean the following case:

 - ci-op gets a lease
 - gets preempted by the kernel or something for 30 minutes
 - comes back and its lease has been reaped?

It would likely fail on the next heartbeat